Enable setting ttl for MockDatastoreService

ytjohn opened the pull request:
This allows people to test their code if they are already setting a ttl. This resolves #4880

I think we would also be fine with not issuing a log warning, but put that in for completeness.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/StackStorm/st2?pullRequest=5468) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/StackStorm/st2?pullRequest=5468) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/StackStorm/st2?pullRequest=5468) it.</sub>

amanda11 reviewed (approved):
This looks good to me, and thanks for the contribution.
Just awaiting the pipelines to run.

ytjohn commented:
@amanda11 Thanks for the approval. I fixed the formatting and updated changelog.  I'm not yet seeing how this change would have impacted the build, so not sure if that's just an ephemeral issue or something that was expecting ttl to be rejected.. I'm now working on setting up a full local development environment for the st2 codebase itself.

armab commented:
Merged, thanks @ytjohn!
